packages/shared/src/hooks/useScrollBottomEvent.ts

```diff
@@ -0,0 +1,27 @@
+import { useLayoutEffect, RefObject, useCallback } from 'react'
+import { debounce } from 'lodash-es'
+
+export function useScrollBottomEvent(ref: RefObject<HTMLDivElement>, cb: () => void) {
```

Comment:
I recommend using https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver.

Reply:
Append an element at the bottom of the container, as it appears trigger the event. Maybe this is a little bit complex than the current implementation.

Reply:
To use IntersectionObserver, we also add a new element at the bottom and observe its distance to the viewport. It's better than watching `scroll` event and calculate the distance by yourself.

You can use this: https://github.com/streamich/react-use/blob/master/docs/useIntersection.md